dev/MRTCore/mrt/mrm/include/mrm/readers/SectionParser.h

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace Microsoft::Resources
+{
+
+/*!
+     * Simple helper class to simplify parsing of mapped file sections.
+     */
+
+template<typename RETURN_TYPE, typename UNIT_TYPE>
+struct SectionParserBase
+{
+protected:
+    _Field_size_bytes_(m_cbData) RETURN_TYPE* m_pData;
+
+    _Field_size_bytes_(m_cbLeft) RETURN_TYPE* m_pNext;
+
+    size_t m_cbData;
+    size_t m_cbLeft;
+
+public:
+    SectionParserBase() : m_pData(nullptr), m_pNext(nullptr), m_cbData(0), m_cbLeft(0) {}
```

Comment:
= default, initialize members on declaration 
(non-blocking)

Reply:
these are legacy codes. we may clean up later.

---
In reply to: [488886527](https://github.com/microsoft/ProjectReunion/pull/185#discussion_r488886527) [](ancestors = 488886527)

Reply:
fine, thanks.